src/common/exponent/exponentHelper.ts

```diff
@@ -0,0 +1,469 @@
+                let modifiedContents = fileHeader + entrypointContents;
+                modifiedContents = modifiedContents.replace(`class ${entrypointComponentName}`, `export class ${entrypointComponentName}`);
+
+                // Prepend ../ to relative imports that refer to something in a parent directory
+                modifiedContents = modifiedContents.replace(/\.\.\/.*/, (substr) => { return `../${substr}`; });
+
+                // Replace ./ with ../ in relative imports that refer to something in the same directory
+                modifiedContents = modifiedContents.replace(/\.\/.*/, (substr) => { return `.${substr}`; });
+
+                return this.fileSystem.writeFile(this.dotvscodePath(EXPONENT_ENTRYPOINT), modifiedContents);
+            }).catch(() => {
+                return Q.reject<void>(new Error("Unable to read entrypoint. Make sure the file exists and it's under the root directory."));
+            });
+    }
+
+    /**
+     * Convert react native project to exponent.
+     * This consists on three steps:
+     * 1. Change the dependency from facebook's react-native to exponent's fork
+     * 2. Create exp.json
+     * 3. Create index and entrypoint for exponent
+     */
+    public configureExponentEnvironment(): Q.Promise<void> {
+        this.lazylyInitialize();
+        Log.logMessage("Making sure your project uses the correct dependencies for exponent. This may take a while...");
+        return this.changeReactNativeToExponent()
+            .then(() => {
+                Log.logMessage("Dependencies are correct. Making sure you have any necesary configuration file.");
+                return this.createExpJson();
+            }).then(() => {
+                Log.logMessage("Project setup is correct. Generating entrypoint code.");
+                return this.createExponentEntrypoint()
+                    .then(() =>
+                        this.createIndex());
+            });
+    }
+
+    /**
+     * Change dependencies to point to original react-native repo
+     */
+    public configureReactNativeEnvironment(): Q.Promise<void> {
+        this.lazylyInitialize();
+        Log.logMessage("Checking react native is correctly setup. This may take a while...");
+        return this.changeExponentToReactNative();
+    }
+
+    /**
+     * Returns the current user. If there is none, asks user for username and password and logins to exponent servers.
+     */
+    public loginToExponent(promptForInformation: (message: string, password: boolean) => Q.Promise<string>, showMessage: (message: string) => Q.Promise<string>): Q.Promise<XDL.IUser> {
+        this.lazylyInitialize();
+        return XDL.currentUser()
+            .then((user) => {
+                if (!user) {
+                    let username = "";
+                    return showMessage("You need to login to exponent. Please provide username and password to login. If you don't have an account we will create one for you.")
```

Comment:
In a couple of weeks exponentjs will change to create the account from their website. Update this when appropriate